model_check.py

```diff
@@ -0,0 +1,758 @@
+        if not os.path.exists(opt.outdir):
+            os.mkdir(opt.outdir)
+
+        config_logging(opt.outdir, opt.verbose, self.short_msid)
+
+        # Store info relevant to processing for use in outputs
+        proc = dict(run_user=os.environ['USER'],
+                    run_time=time.ctime(),
+                    errors=[],
+                    msid=self.msid.upper(),
+                    short_msid=self.short_msid.upper(),
+                    hist_limit=self.hist_limit)
+        proc["msid_limit"] = self.yellow[self.short_msid] - self.margin[self.short_msid]
+        self.logger.info('##############################'
+                    '#######################################')
+        self.logger.info('# %s_check.py run at %s by %s'
+                    % (self.short_msid, proc['run_time'], proc['run_user']))
+        self.logger.info('# %s_check version = %s' % (self.short_msid, self.version))
+        self.logger.info('# model_spec file = %s' % os.path.abspath(opt.model_spec))
+        self.logger.info('###############################'
+                    '######################################\n')
+
+        self.logger.info('Command line options:\n%s\n' % pformat(opt.__dict__))
+
+        # Connect to database (NEED TO USE aca_read)
+        self.logger.info('Connecting to database to get cmd_states')
+        db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read',
+                         database='aca')
+
+        tnow = DateTime(opt.run_start).secs
+        if opt.oflsdir is not None:
+            # Get tstart, tstop, commands from backstop file in opt.oflsdir
+            bs_cmds = self.get_bs_cmds(opt.oflsdir)
+            tstart = bs_cmds[0]['time']
+            tstop = bs_cmds[-1]['time']
+
+            proc.update(dict(datestart=DateTime(tstart).date,
+                             datestop=DateTime(tstop).date))
+        else:
+            tstart = tnow
+
+        # Get temperature telemetry for 3 weeks prior to min(tstart, NOW)
+        telem_msids = [self.msid, 'sim_z', 'dp_pitch', 'dp_dpa_power', 'roll']
+        if self.other_telem is not None:
+            telem_msids += self.other_telem
+        name_map = {'sim_z': 'tscpos', 'dp_pitch': 'pitch'}
+        if self.other_map is not None:
+            name_map.update(self.other_map)
+        tlm = self.get_telem_values(min(tstart, tnow),
+                                    telem_msids,
+                                    days=opt.days,
+                                    name_map=name_map)
+        tlm['tscpos'] *= -397.7225924607
+
+        # make predictions on oflsdir if defined
+        if opt.oflsdir is not None:
+            pred = self.make_week_predict(opt, tstart, tstop, bs_cmds, tlm, db)
+        else:
+            pred = dict(plots=None, viols=None, times=None, states=None,
+                        temps=None)
+
+        # Validation
+        URL = "http://cxc.harvard.edu/mta/ASPECT/%s_daily_check" % self.short_msid
```

Comment:
Sorry to confuse the issue a bit, but having now reminded myself of the details of existing plots in the checkers:
- The validation plots have states vs telemetry for dates prior to the start of the loads (obviously, since they need telemetry).
- The plots @richardjedgar needs should be the states during the loads.
- So it seems the validation plots are not directly of use here and can just be cut entirely.
- New code needed to plot off-nominal roll in this context.